### ISSUE TITLE
Meta duplicate

### ISSUE THREAD

**Issue by thousandsofraccoons** [sveltejs/kit#7493]
### Describe the bug

Meta duplicate is back

### Reproduction

https://github.com/thousandsofraccoons/double-meta

### System Info

```Shell
System:
    OS: Windows 10 10.0.19043
    CPU: (8) x64 Intel(R) Core(TM) i7-10510U CPU @ 1.80GHz
    Memory: 2.39 GB / 15.87 GB
  Binaries:
    Node: 16.0.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.10 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 7.10.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.19041.1266.0), Chromium (107.0.1418.26)
    Internet Explorer: 11.0.19041.1566
  npmPackages:
    @sveltejs/adapter-auto: next => 1.0.0-next.86
    @sveltejs/adapter-vercel: ^1.0.0-next.81 => 1.0.0-next.81
    @sveltejs/kit: next => 1.0.0-next.532
    svelte: ^3.46.0 => 3.52.0
    vite: ^3.1.0 => 3.2.2
```

### Severity

serious, but I can work around it

**Comment by dummdidumm** [comment]
Could you be more specific about what you expect which is wrong? The reproducible without more details/context makes it hard to look into this.

**Comment by thousandsofraccoons** [comment]
@dummdidumm Hello, Simon. I'm not sure why it happens exactly, my guess is because of `svelte-intl-precompile` together with SSR redirect to locale route. 

So basically `<svelte:head>` gets duplicated in such configuration as in the repo.

![image](https://user-images.githubusercontent.com/33694447/199935324-f6442c9d-b238-4e94-9727-305ebfd36b75.png)

**Comment by dummdidumm** [comment]
The problem is the HTML minifier in the `handle` hook which also removes comments. It seems that Svelte uses a header start/end comment to check its integrity. If that comment is removed, the hydration algorithm thinks that header is gone and appends it, duplicating the content.

The quick fix is to set the `removeComments` option to `false`.

@benmccann @tanhauhau I believe you were involved in some of the meta/header fixes, is there anything we can here other than telling people to not remove these comments?

**Comment by thousandsofraccoons** [comment]
@dummdidumm it did the trick, thank you

**Comment by tanhauhau** [comment]
you mean adding something else other than comments to do the trick?

**Comment by dummdidumm** [comment]
probably that would be the solution, yes. I'm not sure though what to replace it with. Some `data-hydrate-head-start/end` attribute on the first and last element? Are `data-` attributes valid on these header tags?

**Comment by benmccann** [comment]
My thought was whether we could add a dev mode warning that would warn no comments were found. I'm not sure if there's always at least one comment, but I suppose we could add a dummy comment if needed

**Comment by dummdidumm** [comment]
So you're stance on this is "the minifier shouldn't remove those comments, and we should warn about it at dev time"?

**Comment by benmccann** [comment]
> So you're stance on this is "the minifier shouldn't remove those comments, and we should warn about it at dev time"?

Yeah. It seems the easiest/safest fix. I think we use comments elsewhere for hydration

**Comment by dummdidumm** [comment]
If you pre-rendered static site ships without any JavaScript (`export const csr = false`) and therefore hydration isn't needed, then you can turn it on.
